frontend/src/metabase/dashboard/constants.js

```diff
@@ -6,3 +8,7 @@ export const SIDEBAR_NAME = {
   sharing: "sharing",
   info: "info",
 };
+
+// most browsers don't use a custom message with beforeunload anymore, just putting here to retain compatibility
```

Comment:
I put this constant here but I'm anticipating that it will be used across the other tasks. Where else can I put this? I was thinking somewhere in `core` but not totally sure

Reply:
I'd use it close to the code usage maybe? if many places are involved, then it's fine

Reply:
Yep, I've moved it into a custom hook that will keep the message in one place with the `react-use` hook, so we can still import the message for tests and then reuse the code without having to import the message everywhere in the code